# Fix building with Visual Studio 2015 Community Edition Update 3: https://bugs.chromium.org/p/chromium/issues/detail?id=627216
# Update GYP SDK paths to point to Windows 10 SDK instead of non-existent Windows 8 SDK directory
# TODO: Remove this patch after switching to GN and VS 2015 Update 3 fix is implemented

--- a/build/toolchain/win/setup_toolchain.py
+++ b/build/toolchain/win/setup_toolchain.py
@@ -123,7 +123,7 @@ def _LoadToolchainEnv(cpu, sdk_dir):
     if not os.path.exists(script_path):
       raise Exception('%s is missing - make sure VC++ tools are installed.' %
                       script_path)
-    args = [script_path, 'amd64_x86' if cpu == 'x86' else 'amd64']
+    args = [script_path, 'amd64_x86' if cpu == 'x86' else 'amd64', '10.0.10586.0']
     variables = _LoadEnvFromBat(args)
   return _ExtractImportantEnvironment(variables)
 
--- a/build/common.gypi
+++ b/build/common.gypi
@@ -5729,7 +5729,7 @@
           'VCLibrarianTool': {
             'AdditionalOptions': ['/ignore:4221'],
             'AdditionalLibraryDirectories': [
-              '<(windows_sdk_path)/Lib/win8/um/x86',
+              '<(windows_sdk_path)/Lib/10.0.10586.0/um/x86',
             ],
           },
           'VCLinkerTool': {
@@ -5746,7 +5746,7 @@
               'shlwapi.lib',
             ],
             'AdditionalLibraryDirectories': [
-              '<(windows_sdk_path)/Lib/win8/um/x86',
+              '<(windows_sdk_path)/Lib/10.0.10586.0/um/x86',
             ],
             'GenerateDebugInformation': 'true',
             'MapFileName': '$(OutDir)\\$(TargetName).map',
